packages/react-dom/src/server/ReactDOMFizzServerBrowser.js

```diff
@@ -32,23 +32,34 @@ type Options = {|
   bootstrapModules?: Array<string>,
   progressiveChunkSize?: number,
   signal?: AbortSignal,
-  onCompleteAll?: () => void,
   onError?: (error: mixed) => void,
 |};
 
+// TODO: Move to sub-classing ReadableStream.
+type ReactDOMServerReadableStream = ReadableStream & {
+  allReady: Promise<void>,
+};
+
 function renderToReadableStream(
   children: ReactNodeList,
   options?: Options,
-): Promise<ReadableStream> {
+): Promise<ReactDOMServerReadableStream> {
   return new Promise((resolve, reject) => {
+    let onCompleteAll;
+    const allReady = new Promise(res => {
```

Comment:
This never rejects which is probably a problem. Should reject if it's a terminal error or aborted I guess.